### PR TITLE
Add Hover Functionality 

### PR DESCRIPTION
**Pull request by AtherBilal**
### PR Safety Checklist:

 - [ ] Added the task to the appropriate release doc under **Enhancements** or **Bug Fixes**
 - [ ] Bump `package.json` & `package-lock.json` version numbers to appropriate release
 - [ ] (optional) All external API changes have been documented
 - [ ] (optional) Build docs: `npm run docs`

### CODE REVIEW

File: src/LayerPanel/LayerPanelLayersPage/index.js [hunk 1]
```diff
@@ -288,8 +329,7 @@ class LayerPanelLayersPage extends Component {
         map.addLayer(layer)
         map.getView().fit(source.getExtent(), map.getSize())
       }
-    }).catch(err => {
-    })
+    }).catch(err => { console.error(err) })
```
Comment (AtherBilal) [inline src/LayerPanel/LayerPanelLayersPage/index.js:332]: replace with `ugh.error`

File: src/LayerPanel/LayerPanelList/index.js [hunk 2]
```diff
@@ -90,13 +90,14 @@ class LayerPanelList extends Component {
     if (children) {
       return (
         <List>
-          <div id='_ol_kit_layer_panel_drag_container' onDragEnd={this.onDragEnd}>
+          <div id='_ol_kit_layer_panel_drag_container' onDragEnd={this.onDragEnd} >
```
Comment (AtherBilal) [inline src/LayerPanel/LayerPanelList/index.js:93]: Remove space

File: src/LayerPanel/LayerPanelList/index.js [hunk 3]
```diff
@@ -90,13 +90,14 @@ class LayerPanelList extends Component {
...
             {React.Children.map(this.props.children, (child, i) => {
+
               const id = `${i}_${nanoid(6)}`
 
               return (
-                <div className={'dropzone'} onDragOver={this.onDragOver} key={id || child.id}>
-                  <div id={id} className={'draggable'} draggable={disableDrag ? false : true} onDragStart={this.onDragStart}>{child}</div>
+                <div className={'dropzone'} onDragOver={this.onDragOver} key={id || child.id} >
+                  <div id={id} className={'draggable'} draggable={disableDrag ? false : true} onDragStart={this.onDragStart} onMouseEnter={() => this.addInteraction} >{child}</div>
```
Comment (AtherBilal) [inline src/LayerPanel/LayerPanelList/index.js:100]: Remove this change

File: src/LayerPanel/LayerPanelList/index.js [hunk 4]
```diff
@@ -108,12 +109,13 @@ class LayerPanelList extends Component {
...
             {items.map((item, i) => {
+              console.log('item: ', item)
               const id = `${i}_${nanoid(6)}`
 
               return (
                 <div className={'dropzone'} onDragOver={this.onDragOver} key={item}>
-                  <div id={id} className={'draggable'} draggable={disableDrag ? false : true} onDragStart={this.onDragStart}>
-                    <LayerPanelListItem>{item}</LayerPanelListItem>
+                  <div id={id} className={'draggable'} draggable={disableDrag ? false : true} onDragStart={this.onDragStart} >
+                    <LayerPanelListItem onMouseOver={() => this.addInteraction}>{item}</LayerPanelListItem>
```
Comment (AtherBilal) [inline src/LayerPanel/LayerPanelList/index.js:118]: Remove

File: src/LayerPanel/LayerPanelList/index.js [hunk 5]
```diff
@@ -108,12 +109,13 @@ class LayerPanelList extends Component {
         <List>
           <div id='_ol_kit_layer_panel_drag_container' onDragEnd={this.onDragEnd}>
             {items.map((item, i) => {
+              console.log('item: ', item)
```
Comment (AtherBilal) [inline src/LayerPanel/LayerPanelList/index.js:112]: Remove log

File: src/Map/utils.js [hunk 6]
```diff
@@ -165,7 +165,7 @@ export function convertXYtoLatLong (map, x, y) {
  * @since 0.2.0
  * @returns {ol.interaction.Select} https://openlayers.org/en/latest/apidoc/module-ol_interaction_Select-Select.html
  */
-export function createSelectInteraction () {
+export function createSelectInteraction (props) {
```
Comment (AtherBilal) [inline src/Map/utils.js:168]: change to `opts`

File: src/Map/Map.js [hunk 7]
```diff
@@ -115,10 +115,11 @@ class Map extends React.Component {
     }
 
     // check the map to see if select interaction has been added
-    const selectInteractionOnMap = map.getInteractions().getArray().find(interaction => {
+    const selectInteractionOnMap = map.getInteractions().getArray()
+      // Layer panel also adds a select interaction
+      .filter(interaction => interaction.type !== '_ol_kit_layer_panel_hover')
```
Comment (AtherBilal) [inline src/Map/Map.js:120]: change to `_ol_kit_interaction_type`

File: src/LayerPanel/LayerPanelLayersPage/index.js [hunk 8]
```diff
@@ -55,6 +59,42 @@ class LayerPanelLayersPage extends Component {
     }
   }
 
+  initializeSelect = map => {
+    const { selectInteraction } = this.props
```
Comment (AtherBilal) [inline src/LayerPanel/LayerPanelLayersPage/index.js:63]: Add custom stroke, fill, and color props

Comment (AtherBilal) [reply src/LayerPanel/LayerPanelLayersPage/index.js:63]: from `setHoverStyle` prop

Comment (AtherBilal) [reply src/LayerPanel/LayerPanelLayersPage/index.js:63]: Default to empty obj `{}`

**Comment by drodenberg** [comment]
has this been updated?

File: src/LayerPanel/LayerPanelLayersPage/index.js [hunk 9]
```diff
@@ -55,6 +60,49 @@ class LayerPanelLayersPage extends Component {
     }
   }
 
+  initializeSelect = map => {
+    const { setHoverStyle, disableHover } = this.props
+
+    if (disableHover) return // opt-out
+
+    const { stroke = 'red', fill = '#ffffff', color = 'red'} = setHoverStyle()
```
Comment (AtherBilal) [inline src/LayerPanel/LayerPanelLayersPage/index.js:68]: `setHoverStyle() || {}`

File: src/LayerPanel/LayerPanelLayersPage/index.js [hunk 10]
```diff
@@ -400,7 +448,9 @@ LayerPanelLayersPage.defaultProps = {
   shouldHideFeatures: (layer) => false,
   shouldAllowLayerRemoval: (layer) => true,
   getMenuItemsForLayer: () => false,
-  tabIcon: <LayersIcon />
+  tabIcon: <LayersIcon />,
+  setHoverStyle: () => ({ color: 'red', fill: '#ffffff', stroke: 'red' }),
```
Comment (AtherBilal) [inline src/LayerPanel/LayerPanelLayersPage/index.js:452]: Remove

Comment (AtherBilal) [reply src/LayerPanel/LayerPanelLayersPage/index.js:452]: Remove function contents

File: src/LayerPanel/LayerPanelLayersPage/index.js (hunk 9 above)
Comment (AtherBilal) [reply src/LayerPanel/LayerPanelLayersPage/index.js:68]: Move to `selectFeatures`

File: src/LayerPanel/LayerPanelLayersPage/index.js [hunk 11]
```diff
@@ -55,6 +60,49 @@ class LayerPanelLayersPage extends Component {
...
+    map.addInteraction(this.selectInteraction)
+  }
+
+  selectFeatures = features => {
+    const { disableHover } = this.props
+
+    if (disableHover) return
+    // clear the previously selected feature before adding newly selected feature so only one feature is "selected" at a time
+    this.selectInteraction.getFeatures().clear()
+    features.forEach(feature => {
```
Comment (AtherBilal) [inline src/LayerPanel/LayerPanelLayersPage/index.js:99]: Check if interaction style can be updated here

**Review by drodenberg — APPROVED** [review]

**Pull request merged**